service/handler.go

```diff
@@ -227,6 +227,13 @@ func (h *HandlerService) processPacket(a agent.Agent, p *packet.Packet) error {
 			return fmt.Errorf("Invalid handshake data. Id=%d", a.GetSession().ID())
 		}
 
+		for _, fun := range a.GetSession().GetHandshakeValidators() {
+			if err := fun(handshakeData); err != nil {
```

Comment:
The way it is, an error is returned when the first validation fails, ignoring following validations. Would it be interesting to run all validators and return an error if any one of them fails? This would provide less back and forth in case multiple validations are to fail.

Reply:
To be honest, I'm not sure if this use-case if very valuable, though.

Reply:
For simplicity sake's I say lets keep the way it is right now. The end result is the same whether multiple or only one validator failed.